Extensions/ArtifactEngine/Providers/cacheProvider.ts

```diff
@@ -0,0 +1,103 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as readline from 'readline';
+import { resolve } from 'url';;
+var crypto = require('crypto');
+
+var tl = require('vsts-task-lib/task');
+
+import * as models from '../Models';
+import { ArtifactItemStore } from '../Store/artifactItemStore';
+import { Logger } from '../Engine/logger';
+
+export class CacheProvider implements models.IArtifactProvider {
+
+    public artifactItemStore: ArtifactItemStore;
+
+    constructor(artifactCacheDirectory: string, artifactCacheKey: string, relPath?: string) {
+        this.artifactCacheDirectory = artifactCacheDirectory;
+        this.key = crypto.createHash('SHA256').update(artifactCacheKey).digest('hex');
+        this.relPath = relPath ? relPath : '';
+        this.cacheDirectory = path.join(artifactCacheDirectory, models.Constants.CacheFolder, this.key);
+        this.cacheHashPath = path.join(this.cacheDirectory, 'artifact-metadata.csv');
+        this.makeOldHash()
+    }
+
+    getRootItems(): Promise<models.ArtifactItem[]> {
+        throw new Error("Not implemented");
+    }
+
+    getArtifactItems(artifactItem: models.ArtifactItem): Promise<models.ArtifactItem[]> {
+        throw new Error("Not implemented");
+    }
+
+    getArtifactItem(artifactItem: models.ArtifactItem): Promise<NodeJS.ReadableStream> {
+        var promise = new Promise<NodeJS.ReadableStream>((resolve, reject) => {
+            if (artifactItem.fileHash && artifactItem.fileHash === this.oldFileHashMap[path.normalize(artifactItem.path.substring(this.relPath.length + 1))]
+                && fs.existsSync(path.join(this.artifactCacheDirectory, models.Constants.CacheFolder, this.key, path.normalize(artifactItem.path.substring(this.relPath.length + 1))))) {
+                const inputStream = fs.createReadStream(path.join(this.artifactCacheDirectory, models.Constants.CacheFolder, this.key, artifactItem.path.substring(this.relPath.length + 1)));
```

Comment:
no need for path.normalize here? if so, can you store this in a variable and use it across both places?